results/client.go

```diff
@@ -114,159 +132,93 @@ func (r *Archive) WalkFilesWithPrefix(prefix string, walkfn filepath.WalkFunc) e
 		if err != nil {
 			return err
 		}
-
-		if strings.HasPrefix(filepath.Clean(header.Name), prefix) {
-			info := &tarFileInfo{
-				header.FileInfo(),
-				tr,
-			}
-			err = walkfn(filepath.Clean(header.Name), info, err)
+		info := &tarFileInfo{
+			header.FileInfo(),
+			tr,
 		}
+		err = walkfn(filepath.Clean(header.Name), info, err)
 	}
 	return nil
-}
-
-// NamespacedResources iterates and executes walkfn for every namespaced resource file.
-func (r *Archive) NamespacedResources(walkfn filepath.WalkFunc) error {
-	return r.WalkFilesWithPrefix(NamespacedResourcesDir, walkfn)
-}
-
-// NonNamespacedResources iterates and executes walkfn for every non-namespaced resource file.
-func (r *Archive) NonNamespacedResources(walkfn filepath.WalkFunc) error {
-	return r.WalkFilesWithPrefix(nonNamespacedResources(r.Version), walkfn)
-}
 
-// Hosts iterates and executes walkfn for all files relating to the host data.
-func (r *Archive) Hosts(walkfn filepath.WalkFunc) error {
-	return r.WalkFilesWithPrefix(HostsDir, walkfn)
 }
 
-// Plugins walks each plugin file executing walkfn on each.
-func (r *Archive) Plugins(walkfn filepath.WalkFunc) error {
-	return r.WalkFilesWithPrefix(PluginsDir, walkfn)
-}
+// Functions to be used within a walkfn
 
-// PodLogs walks each pod log file executing walkfn on each.
-func (r *Archive) PodLogs(walkfn filepath.WalkFunc) error {
-	return r.WalkFilesWithPrefix(PodLogs, walkfn)
-}
+// ExtractFileIntoStruct takes data from the `file` and decodes the data into the object.
+func (a *Archive) ExtractFileIntoStruct(file, path string, info os.FileInfo, object interface{}) error {
```

Comment:
Also considering changing the `file` parameter to a predicate function instead. Then `ExtractConfig` can go away.